assets/js/src/form_editor/blocks/custom_text/custom_field_settings.jsx

```diff
@@ -0,0 +1,93 @@
+import React, { useState } from 'react';
+import {
+  Button,
+  SelectControl,
+  ToggleControl,
+} from '@wordpress/components';
+import PropTypes from 'prop-types';
+import MailPoet from 'mailpoet';
+import { useDispatch, useSelect } from '@wordpress/data';
+
+const CustomFieldSettings = ({
+  mandatory,
+  validate,
+  customFieldId,
+  updateAttributes,
+}) => {
+  const [localMandatory, setLocalMandatory] = useState(mandatory);
+  const [localValidate, setLocalValidate] = useState(validate);
+  const { saveCustomField } = useDispatch('mailpoet-form-editor');
+  const isSaving = useSelect(
+    (sel) => sel('mailpoet-form-editor').getIsCustomFieldSaving(),
+    []
+  );
```

Comment:
Maybe we should pass those two using props so that this component is reusable for user import etc.